modules/ghlint-extensions/src/test/kotlin/net/twisterrob/ghlint/model/IsCheckoutKtTest.kt

```diff
@@ -0,0 +1,136 @@
+package net.twisterrob.ghlint.model
+
+import io.kotest.matchers.shouldBe
+import net.twisterrob.ghlint.testing.load
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+
+class IsCheckoutKtTest {
+
+	private fun loadFirstStep(@Language("yaml") yaml: String, fileName: String = "test.yml"): Step {
+		val file = load(yaml, fileName)
+		return when (val content = file.content) {
+			is Workflow -> (content.jobs.values.single() as Job.NormalJob).steps.single()
+			is Action -> (content.runs as Action.Runs.CompositeRuns).steps.single()
+			is InvalidContent -> error("Invalid content: ${content.error}")
+		}
+	}
+
+	@MethodSource("getCheckoutActions")
+	@ParameterizedTest fun `standard checkout actions in workflow`(action: String) {
+		val step = loadFirstStep(
+			"""
+				on: push
+				jobs:
+				  test:
+				    runs-on: test
+				    steps:
+				      - uses: ${action}
+			""".trimIndent(),
+		)
+
+		step.isCheckout shouldBe true
+	}
```

Comment:
@timsearle you were asking how to create pieces of the workflow for test, I think this is the way. Obviously it's overkill in this isCheckout, but it's very useful in seesEnvVar's case.